src/test/java/com/puppycrawl/tools/checkstyle/internal/XdocsPagesTest.java

```diff
@@ -973,98 +973,55 @@ private static String getModulePropertyExpectedTypeName(String sectionName, Clas
-                result += " (either the empty string or an uppercase ISO 3166 2-letter code)";
-            }
-            else if ("Checker".equals(sectionName) && "localeLanguage".equals(propertyName)) {
-                result += " (either the empty string or a lowercase ISO 639 code)";
-            }
+            result = propertyTypeNameForString(sectionName, propertyName);
         }
         else if (fieldClass == String[].class) {
-            if (propertyName.endsWith("Tokens") || propertyName.endsWith("Token")
-                    || "AtclauseOrderCheck".equals(instanceName) && "target".equals(propertyName)
-                    || "MultipleStringLiteralsCheck".equals(instanceName)
-                            && "ignoreOccurrenceContext".equals(propertyName)) {
-                result = "subset of tokens TokenTypes";
-            }
-            else {
-                result = "String[]";
-            }
-        }
-        else if (fieldClass == URI.class) {
-            result = "URI";
+            result = propertyTypeNameForStringArray(propertyName, instanceName);
         }
         else if (fieldClass == Pattern.class) {
-            if ("SuppressionSingleFilter:checks".equals(checkProperty)
-                || "SuppressionXpathSingleFilter:files".equals(checkProperty)
-                || "SuppressionXpathSingleFilter:checks".equals(checkProperty)
-                || "SuppressionXpathSingleFilter:message".equals(checkProperty)
-                || "IllegalTokenText:format".equals(checkProperty)) {
-                result = "Regular Expression";
-            }
-            else {
-                result = "Pattern";
-            }
+            result = propertyTypeNameForPattern(checkProperty);
         }
         else if (fieldClass == Pattern[].class) {
-            if ("ImportOrder:groups".equals(checkProperty)
-                || "ImportOrder:staticGroups".equals(checkProperty)
-                || "ClassDataAbstractionCoupling:excludeClassesRegexps".equals(checkProperty)
-                || "ClassFanOutComplexity:excludeClassesRegexps".equals(checkProperty)) {
-                result = "Regular Expressions";
-            }
-            else {
-                result = "Pattern[]";
-            }
+            result = propertyTypeNameForPatternArray(checkProperty);
         }
         else if (fieldClass == Scope.class) {
             result = "Scope";
         }
-        else if (fieldClass == AccessModifierOption[].class) {
-            result = "AccessModifierOption[]";
-        }
         else if ("PropertyCacheFile".equals(fieldClass.getSimpleName())) {
             result = "File";
         }
         else if (fieldClass.isEnum()) {
             result = fieldClass.getSimpleName();
         }
         else {
-            fail("Unknown property type: " + fieldClass.getSimpleName());
+            switch (fieldClass.getSimpleName()) {
+                case "int":
```

Comment:
Too bad that it is not possible to use `switch(fieldClass)`.
I would like to avoid calling "getSimpleName" here.

Please add a method like
```java
boolean isKnownPropertyType(Class<?> propertyType) {
  return fieldClass == int.class
      || fieldClass == double[].class
      || ....
}
```
to use it here
```java
else if (isKnownPropertyType(fieldClass)) {
  result = fieldClass.getSimpleName();
}
else {
  fail("Unknown property type: " + fieldClass.getSimpleName());
}
```

Reply:
Done